### DISCUSSION TITLE
Scrollbar

### DISCUSSION THREAD

**Discussion by sioodmy** [YaLTeR/niri#482] [Ideas]
![image](https://github.com/YaLTeR/niri/assets/81568712/ccee3ef0-1034-42c9-8597-8dce07339164)

I want to add a slider to my bar indicating how wide current workspace(if that makes any sense)
just like a vertical scrollbar 

it would be nice to have some way of getting workspace width and current position through niri msg

**Comment by YaLTeR** [comment]
I guess this is https://github.com/YaLTeR/niri/pull/453 content; however, I'm not entirely sold on signaling *continuous* values through IPC since they will change very frequently, e.g. during a gesture with a 1000 Hz mouse you're looking at a new value every 1 ms. Maybe we could signal this only when the view is stationary, i.e. no gesture or animation running?

**Reply by alexindigo** [reply]
@YaLTeR how about sending current window number and number of all windows, and windows' horizontal sizes, that way some semblance of scrolling could be recreated without need to send new event every millisecond?

**Reply by alexindigo** [reply]
I got here by searching for scrollbar for niri, just trying to understand where I am within current workspace.

**Reply by YaLTeR** [reply]
> sending current window number and number of all windows, and windows' horizontal sizes

Niri already provides most of this information through IPC: https://yalter.github.io/niri/niri_ipc/struct.Window.html#structfield.layout

The current view position is missing though, which means we're back at my original comment. Another problem is that you'd have to compute window positions exactly how niri does it with gaps and whatnot, that's quite error-prone.

**Reply by alexindigo** [reply]
Thanks for the quick response.

I'm curious why gaps needed to be accounted for, since they all uniform, they can be ignored.
In my understanding, all I need to know that current window is 42% from the left edge, and takes 8% of the whole roll, and there is 50% more to the right. Does it make sense?

**Comment by norveganon** [comment]
Not the same, but I find this script quite useful to indicate the amount of columns I have in a workspace as well as which one is currently focused:
https://git.ficd.sh/ficd/niri-scripts/src/branch/main/windows

It doesn't give you an exact %, but it gives you an idea about where you are and what you got to deal with.

**Reply by alexindigo** [reply]
Thank you, will check it out.
